stores/metadata.go

```diff
@@ -666,18 +673,32 @@ FROM (
 	FROM (
 		SELECT size, SUBSTR(object_id, ?) AS trimmed
 		FROM objects
-		WHERE object_id LIKE ?
+		WHERE ?
 	) AS i
 ) AS m
 GROUP BY name
-HAVING name LIKE ? AND name != ?
-LIMIT ? OFFSET ?`, sqlConcat(s.db, "?", "trimmed"), sqlConcat(s.db, "?", "substr(trimmed, 1, slashindex)")), path, path, utf8.RuneCountInString(path)+1, path+"%", fmt.Sprintf("%s%s", path, prefix+"%"), path, limit, offset)
+HAVING ? AND name != ?
+LIMIT ? OFFSET ?`,
+		sqlConcat(s.db, "?", "trimmed"),
+		sqlConcat(s.db, "?", "substr(trimmed, 1, slashindex)")),
+		path,
+		path,
+		utf8.RuneCountInString(path)+1,
+		sqlHasPrefix("object_id", path),
+		sqlHasPrefix("name", path+prefix),
+		path,
+		limit,
+		offset)
 
 	var metadata []api.ObjectMetadata
 	err := query.Scan(&metadata).Error
 	if err != nil {
 		return nil, err
 	}
+	// Sort results outside of query to get consistent ordering across dbs.
+	sort.Slice(metadata, func(i, j int) bool {
```

Comment:
not sure what the problem was exactly but wouldn't `ORDER BY LOWER(name)` have worked?

Reply:
If we were only talking ASCII characters then probably yes but collations are weird. Even though both MySQL and SQLite use a binary collation for the field they ended up returning different orders for paths that were lowercase to begin with but started with `/` or `//` for example.